clang/lib/Driver/Driver.cpp

```diff
@@ -131,8 +134,22 @@ static bool shouldRaiseHost(const ArgList &Args) {
     constexpr bool Default = CLANG_ENABLE_OPAQUE_POINTERS_INTERNAL;
     return (Iter == FilteredArgs.end()) ? Default : Pos == (*Iter)->getValue();
   };
-  return !Args.hasArg(options::OPT_fsyntax_only) &&
-         Args.hasArg(options::OPT_fsycl_raise_host) && UseOpaquePointers();
+  bool ShouldRaise = Args.hasArg(options::OPT_fsycl_raise_host);
+  if (ShouldRaise) {
+    if (Args.hasArg(options::OPT_fsyntax_only))
+      ShouldRaise = false;
```

Comment:
Should we use early returns here, in accordance with the [LLVM coding standards](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)?

Reply:
no, that would prevent the diagnosis of other flags.